### ISSUE TITLE
Add workspace when starting a pipeline or task

### ISSUE THREAD

**Issue by siamaksade** [tektoncd/cli#651]
Given the workspace support in pipelines, currently one has to create a `PipelineRun` yaml in order to start a pipeline that requires a workspace. The `tkn pipeline start` command should be augmented to be able to map persistent volume claims to workspace when starting a pipeline.

Suggestion:
```
tkn pipeline start -w|--workspace my-workspace=my-workspace-pvc
```

The above syntax is similar to pipeline resources

**Comment by chmouel** [comment]
maybe we can have a workspace backed by configmap  if no pvc is specified?

**Comment by sbwsg** [comment]
We also are discussing having an "auto" workspace just in case the user doesn't want / know how to provide one.  This would be someone the pipeline controller injects in response to some config and would work similarly to how pipeline resources have a PVC generated for them today (called the "artifacts pvc"), but would be more transparent to the user (hopefully).

**Comment by siamaksade** [comment]
Auto workspace is very useful. Still makes sense to allow users to override it but it would definitely simplify starting pipelines.

**Comment by vdemeester** [comment]
It could be something like:

- `--workspace my-workspace=pvc/…` 
- `--workspace my-workspace=emptyDir`
- `--workspace my-workspace=configmap/…`

**Comment by piyush-garg** [comment]
Closed by #826 and #854
